pkg/watch/mux.go

```diff
@@ -149,6 +151,7 @@ func (m *Broadcaster) Action(action EventType, obj runtime.Object) {
 // You must not call Action after calling Shutdown.
```

Comment:
I had not noticed that our channels could be buffered, you're right.

I can't find any reference to this, but I am guessing that if you close a buffered channel, it can still be read from and its buffered values are not lost.